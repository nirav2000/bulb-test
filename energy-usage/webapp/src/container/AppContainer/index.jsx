import React from 'react';
import Axios from 'axios';
import Chart from '../../components/Chart';
import Table from '../../components/Table';
import { getEstimateEnergyUsage } from '../../utils/EnergyUsageConverter';

class AppContainer extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      energyUsageData: [],
      meterReadings: [],
      isDataLoaded: false,
      isDataError: false,
    };
  }

  componentDidMount() {
    Axios.get('https://storage.googleapis.com/bulb-interview/meterReadingsReal.json')
      .then(res => this.setState({
        meterReadings: res.data.electricity,
        energyUsageData: getEstimateEnergyUsage(res.data.electricity),
        isDataLoaded: res !== undefined,
      }))
      .catch(err => this.setState({ isDataError: err !== undefined }));
  }

  render() {
    const {
      meterReadings, energyUsageData, isDataLoaded, isDataError,
    } = this.state;
    if (isDataLoaded) {
      return (
        <div >
          <Chart energyUsageData={energyUsageData} />
          <Table meterReadings={meterReadings} />
        </div >
      );
    }

    if (isDataError) {
      return <div>There is a problem loading the data, please try again</div>;
    }

    return (
      <div>Loading data...</div>
    );
  }
}

export default AppContainer;
