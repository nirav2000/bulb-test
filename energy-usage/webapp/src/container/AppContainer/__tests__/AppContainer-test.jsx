import React from 'react';
import sinon from 'sinon';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import AppContainer from '../index';

describe('<AppContainer /> test', () => {
  it('Test snapshots of content', () => {
    const wrapper = shallow(<AppContainer />);

    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });

  it('should call componentDidMount', () => {
    sinon.spy(AppContainer.prototype, 'componentDidMount');
    mount(<AppContainer />);

    expect(AppContainer.prototype.componentDidMount).toHaveProperty('callCount', 1);
    AppContainer.prototype.componentDidMount.restore();
  });
});
