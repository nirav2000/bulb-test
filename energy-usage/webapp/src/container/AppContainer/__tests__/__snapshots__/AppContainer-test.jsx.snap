// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppContainer /> test Test snapshots of content 1`] = `
<div>
  Loading data...
</div>
`;
